### ISSUE TITLE
terminal does not link class files from current project or class files from libraries we depend on

### ISSUE THREAD

**Issue by jurgenvinju** [cwi-swat/rascal-vscode#37]
see subject. the result is that bespoke Java code which is called from Rascal code throws exceptions about missing classes.

this has to be solved via the `classloaders` parameter of pathConfig and using that to add classloaders to the interpreter's configuration.

**Comment by jurgenvinju** [comment]
also depends on #38

**Comment by jurgenvinju** [comment]
I've added the classloaders but the next bug appeared:

```
rascal>import lang::flybytes::demo::pico::Compiler;
INFO: resolved |lib://rascal-lsp| at |jar+file:///Users/jurgenv/git/rascal-vscode/rascal-vscode-extension/out/rascal-lsp.jar!/|
INFO: resolved |lib://rascal| at |jar+file:///Users/jurgenv/git/rascal-vscode/rascal-vscode-extension/out/rascal.jar!/|
org.rascalmpl.exceptions.JavaMethodLink: Cannot link method lang.flybytes.internal.ClassCompiler because: IValueFactory
        at org.rascalmpl.interpreter.utils.JavaBridge.getJavaClassInstance(JavaBridge.java:400)
        at org.rascalmpl.interpreter.result.JavaMethod.<init>(JavaMethod.java:95)
        at org.rascalmpl.interpreter.result.JavaMethod.<init>(JavaMethod.java:60)
        at org.rascalmpl.semantics.dynamic.FunctionDeclaration$Abstract.interpret(FunctionDeclaration.java:61)
        at org.rascalmpl.semantics.dynamic.Declaration$Function.interpret(Declaration.java:117)
        at org.rascalmpl.semantics.dynamic.Toplevel$GivenVisibility.interpret(Toplevel.java:35)
```

**Comment by jurgenvinju** [comment]
It seems that even though the IValueFactory.class.getClassLoader() is added, the JavaBridge does not use a classloader which effectively integrates the classloaders into a single name space (such as URLClassLoader does with a list of URLs)

**Comment by jurgenvinju** [comment]
the problem is caused by the SourceLocationClassLoader which is offered only the locations of the additional libraries but not the locations of the rascal standard library and the rascal-lsp library. So when this classloader defers to its parent for finding IValueFactory, it goes to the System classloader which only sees the jar that the LSPTerminal class is in.

**Comment by jurgenvinju** [comment]
this is an issue with the normal `RascalShell` as well. It throws the exact same exceptions when we start a terminal on the commandline for the `flybytes` project and `import lang::flybytes::Compiler`. This makes it easier to debug. If we fix it for `RascalShell` the same fix would apply for `LSPTerminalREPL`

**Comment by jurgenvinju** [comment]
closed via usethesource/rascal#1487
